cmd/ddev/cmd/exec_test.go

```diff
@@ -99,13 +99,10 @@ func TestCmdExec(t *testing.T) {
 
 	assert.FileExists(filepath.Join(site.Dir, "TestCmdExec-touch-all-in-one.txt"))
 
-	_, err = exec.RunHostCommand(DdevBin, "exec", "true", "&&", "touch", "/var/www/html/TestCmdExec-touch-separate-args.txt")
+	// Complex commands with spaces should also work, even without the --raw flag
+	out, err = exec.RunHostCommand(DdevBin, "exec", "bash", "-c", "for i; do echo $i; done", "_", "string with spaces")
 	assert.NoError(err)
-
-	err = app.MutagenSyncFlush()
-	assert.NoError(err)
-
-	assert.FileExists(filepath.Join(site.Dir, "TestCmdExec-touch-separate-args.txt"))
```

Comment:
This check for `TestCmdExec-touch-separate-args.txt` fails with new logic for `ddev exec`.

```
ddev exec true && touch /var/www/html/TestCmdExec-touch-separate-args.txt
```

In bash, this is done like this, so `touch` isn't even a part of `ddev exec`:

```
( ddev exec true ) && ( touch /var/www/html/TestCmdExec-touch-separate-args.txt )
```

But in v1.24.2 `exec.RunHostCommand`, it was executed like this:

```
ddev exec ( true && touch /var/www/html/TestCmdExec-touch-separate-args.txt )
```

I don't understand why this check even works with v1.24.2.

Anyway, in this PR it works the same way as it does in bash, it fails because there is file created.

I don't see the point in checking this condition, so I replaced it with another check.

Reply:
We have a bit of a misunderstanding of this complicated stuff here. This worked fine in HEAD, and can be demonstrated and improved via:
```
	out, err = exec.RunHostCommand(DdevBin, "exec", "true", "&&", "touch", "/var/www/html/TestCmdExec-touch-separate-args.txt", "&&", "hostname")
	assert.NoError(err)
	out = strings.Trim(out, "\r\n ")
	assert.Equal(fmt.Sprintf("%s-web", site.Name), out)
```

In this situation, all of the args to RunHostCommand() are arguments, and they're not intercepted by bash on the host side because we're not using bash in RunHostCommand(). 

It's so very hard dealing with multiple bash interpretations. It's possible that we need some aspects of this test that do RunHostCommand using `bash -c` instead of just using DdevBin as the interpreter.

Reply:
(Taking this PR fairly slowly because of the difficulty of the terrain and the number of times we've introduced bugs that were hard to sort out in specific situations.)

Reply:
I added two test cases to see how it works in Bash:

```
bash -c "ddev exec 'true && pwd'"
bash -c "ddev exec true && pwd"
```

I don't check for file here, because it would be not reasonable to create a `/var/www/html/TestCmdExec-touch-separate-args.txt` file on the host.

Reply:
> because it would be not reasonable to create a /var/www/html/TestCmdExec-touch-separate-args.txt file on the host

And /var/www/html doesn't exist on the host in (most) cases and where it does, needs permissions of course.